SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityJobPoller.java

```diff
@@ -69,6 +72,16 @@ public void runActionOnPoll() {
         continue;
       }
 
+      if (start - taskId.getStartedAt() >= configuration.getTaskExecutionTimeout()) {
```

Comment:
we also need to be checking the Request type. As-is, this would kill a web service after running for 12 hours, which isn't what we want -- we should only be checking this for non-long running Requests (i.e. cron jobs, on-demand and run once tasks)